homeassistant/components/shelly/event.py

```diff
@@ -0,0 +1,107 @@
+"""Event for Shelly."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Final
+
+from homeassistant.components.event import (
+    DOMAIN as EVENT_DOMAIN,
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import RPC_INPUTS_EVENTS_TYPES
+from .coordinator import ShellyRpcCoordinator, get_entry_data
+from .utils import (
+    async_remove_shelly_entity,
+    get_device_entry_gen,
+    get_rpc_input_name,
+    get_rpc_key_instances,
+    is_rpc_momentary_input,
+)
+
+
+@dataclass
+class ShellyEventDescription(EventEntityDescription):
+    """Class to describe Shelly event."""
+
+    removal_condition: Callable[[dict, dict, str], bool] | None = None
+
+
+RPC_EVENT: Final = ShellyEventDescription(
+    key="input",
+    device_class=EventDeviceClass.BUTTON,
+    event_types=list(RPC_INPUTS_EVENTS_TYPES),
```

Comment:
I'm missing translations for the entity event types.

Reply:
Oops I totally forgot about that. I'll open a new PR with translations.

Reply:
Done https://github.com/home-assistant/core/pull/100608